sklearn/preprocessing/data.py

```diff
@@ -138,8 +139,39 @@ def scale(X, axis=0, with_mean=True, with_std=True, copy=True):
         Xr = np.rollaxis(X, axis)
         if with_mean:
             Xr -= mean_
+            mean_1 = Xr.mean(axis=0)
+            # Verify that mean_1 is 'close to zero'. If X contains very
+            # large values, mean_1 can also be very large, due to a lack of
+            # precision of mean_. In this case, a pre-scaling of the
+            # concerned feature is efficient, for instance by its mean or
+            # maximum.
+            if isinstance(mean_1, np.ndarray):
+                if not np.allclose(mean_1, np.zeros(np.shape(mean_1)[0])):
+                    raise ValueError(
```

Comment:
Yes but in this case we fail to return a centered result. Actually the obtained X_r is false and according to me it is then useless to keep running the program... isn't it ?